htdocs/accountancy/class/accountancyimport.class.php

```diff
@@ -73,17 +108,24 @@ public function computeAmount(&$arrayrecord, $fieldname, &$listfields, &$listval
 			$fieldname = $fieldArr[1];
 		}
 
-		$debit  = floatval(trim($arrayrecord[11]['val']));
-		$credit = floatval(trim($arrayrecord[12]['val']));
-		if (!empty($debit)) {
-			$amount = $debit;
-		} else {
-			$amount = $credit;
+		// get fields indexes
+		$field_index_list = array_flip($listfields);
+		if (isset($field_index_list['debit']) && isset($field_index_list['credit'])) {
+			$debit_index = $field_index_list['debit'];
+			$credit_index = $field_index_list['credit'];
+
+			$debit  = floatval($arrayrecord[$debit_index]['val']);
+			$credit = floatval($arrayrecord[$credit_index]['val']);
+			if (!empty($debit)) {
+				$amount = $debit;
+			} else {
+				$amount = $credit;
+			}
+
+			$listfields[] = $fieldname;
```

Comment:
You should not modify content of $listfields and $listvalues. Only the $arrayrecord that is the array that will be used for output should be modified.
Modification of a value should be done into the $arrayrecord directly. You should know which entry to modify by using the ($key - 1) parameter that is send when you call
`$res = call_user_func_array(array($classinstance, $method), array(&$arrayrecord, $fieldname, $listfields, $listvalues, $key - 1));`
but that is missing in parameters of computeAmount().